src/Microsoft.ML.Core/Data/IDataView.cs

```diff
@@ -85,10 +86,10 @@ public interface IDataView
         /// <summary>
         /// Get a row cursor. The active column indices are those for which needCol(col) returns true.
         /// The schema of the returned cursor will be the same as the schema of the IDataView, but getting
-        /// a getter for an inactive columns will throw. The <paramref name="needCol"/> predicate must be
-        /// non-null. To activate all columns, pass "col => true".
+        /// a getter for inactive columns will throw. The <paramref name="colsNeeded"/> indicate the columns that are needed
+        /// to iterate over.If set to an empty <see cref="IEnumerable"/> no column is requested.
         /// </summary>
-        RowCursor GetRowCursor(Func<int, bool> needCol, Random rand = null);
+        RowCursor GetRowCursor(IEnumerable<Schema.Column> colsNeeded, Random rand = null);
```

Comment:
Can you spell out the parameter name? `colsNeeded` => `columnsNeeded`.

[https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/general-naming-conventions](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/general-naming-conventions)

X DO NOT use abbreviations or contractions as part of identifier names.

For example, use GetWindow rather than GetWin.